### PR TITLE
Adds rationale and notes around comma-separated HTTP headers

### PR DESCRIPTION
**Pull request by adriancole**

### CODE REVIEW

File: brave/RATIONALE.md [hunk 1]
```diff
@@ -60,6 +60,15 @@ attempting to do that would limit the applicability of Brave, which is an anti-
 goal. Instead, we choose to use nothing except floor Java version features,
 currently Java 6.
 
+Here's an example of when things that seem right aren't. We once dropped our
```
Comment (adriancole) [inline brave/RATIONALE.md:63]: this is not about HTTP, but don't feel like moving it to another PR ...

File: brave/RATIONALE.md [hunk 2]
```diff
@@ -170,6 +179,74 @@ direct integration with request types (such as HTTP request) vs routing
...
+messaging and RPC where multi-map fields are unusual.
+
+HTTP [RFC 7230](https://tools.ietf.org/html/rfc7230#section-3.2.2) allows encoding of multiple fields with the same name, but this
+only applies to fields which define themselves as comma-separated. The `Setter`
+abstraction has no knowledge of if the request is an HTTP request, and even if
+it did, it wouldn't have knowledge of the format rules per field. That's the
+responsibility of the `Propagation` implementation. Hence, the `Injector`,
+which knows format rules, should do comma joining when valid.
+
+Some think that sending a header in comma-separated format, implies a receiver
```
Comment (adriancole) [inline brave/RATIONALE.md:208]: I added this point because it is a common fantasy

File: brave/RATIONALE.md [hunk 3]
```diff
@@ -60,6 +60,15 @@ attempting to do that would limit the applicability of Brave, which is an anti-
 goal. Instead, we choose to use nothing except floor Java version features,
 currently Java 6.
 
+Here's an example of when things that seem right aren't. We once dropped our
+internal `@Nullable` annotation (which is source retention), in favor of JSR
+305 (which is runtime retention). In doing so, we got `null` analysis from
```
Comment (NersesAM) [inline brave/RATIONALE.md:65]: you can still get `null` analysis from intellij if you add your custom annotation here 
image

Comment (adriancole) [reply brave/RATIONALE.md:65]: Thanks @NersesAM. I just updated with this.

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -6,6 +6,11 @@ send a pull request (on a branch other than `master` or `gh-pages`).
 When submitting code, please apply [Square Code Style](https://github.com/square/java-code-styles).
 * If the settings import correctly, CodeStyle/Java will be named Square and use 2 space tab and indent, with 4 space continuation indent.
 
+You may also want to configure your IDE to perform `null` analysis using Brave's
+source retention annotation `brave.internal.Nullable`. For example, in IntelliJ,
+search for "Nullable" under "Inspections". Click "Configure Annotatione", then
```
Comment (NersesAM) [inline .github/CONTRIBUTING.md:11]: typo in `Annotatione`

Comment (adriancole) [inline .github/CONTRIBUTING.md:11]:
```suggestion
search for "Nullable" under "Inspections". Click "Configure Annotations", then
```

Comment (adriancole) [reply .github/CONTRIBUTING.md]: thx

**Comment by adriancole** [comment]
Thanks for the help @NersesAM !
